src/generator/AutoRest.Go/Model/CodeModelGo.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core;
+using AutoRest.Core.Model;
+using AutoRest.Core.Utilities;
+
+namespace AutoRest.Go.Model
+{
+    public class CodeModelGo : CodeModel
+    {
+        public CodeModelGo()
+        {
+            NextMethodUndefined = new List<IModelType>();
+            PagedTypes = new Dictionary<IModelType, string>();
+        }
+
+        public string[] Version => CodeNamerGo.SDKVersionFromPackageVersion(
+                                            !string.IsNullOrEmpty(Settings.Instance.PackageVersion)
+                                                    ? Settings.Instance.PackageVersion
+                                                    : "0.0.0");
+
+        public override string Namespace
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(base.Namespace))
+                    return base.Namespace;
+
+                return base.Namespace.ToLowerInvariant();
+            }
+            set
+            {
+                base.Namespace = value;
+            }
+        }
+
+        public string ServiceName
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(Namespace))
+                {
+                    return CodeNamerGo.Instance.PascalCase(Namespace);
+                }
+                return string.Empty;
+            }
+        }
+
+        public string GetDocumentation()
+        {
+            return string.Format("Package {0} implements the Azure ARM {1} service API version {2}.\n\n{3}", Namespace, ServiceName, ApiVersion,
```

Comment:
You should probably consider moving to interpolated strings :

public string DocumentationText => $"Package {Namespace} implements the Azure ARM {ServiceName} service API version {ApiVersion}.\n\n{(Documentation??string.Empty) .UnrwapAnchorTags()}"

String.Format isn't very fast, subject to typos in parameter ordering, and is kinda wordy :)